sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -36,7 +36,7 @@ case class PrecisionInfo(precision: Int, scale: Int) {
 /**
  * :: DeveloperApi ::
  * The data type representing `java.math.BigDecimal` values.
- * A Decimal that might have fixed precision and scale, or unlimited values for these.
+ * A Decimal that must have fixed precision and scale.
  *
  * Please use [[DataTypes.createDecimalType()]] to create a specific instance.
  */
```

Comment:
after looking through the code - it definitely makes sense to just remove the Option, and make precision/scale the two arguments on DecimalType.